### PR TITLE
feat: [CDS-86300]: Added the endpoint to update multiple version in git

### PR DESCRIPTION
**Pull request by shivamnegi94**
## Describe your changes
Added the Endpoint to update multiple Git files for template.

`curl --location --request POST 'https://localhost:8181/template/api/templates/update/git-metadata/TestTemp1234?accountIdentifier=kmpySmUISimoRrJL6NL73w&orgIdentifier=default&projectIdentifier=TestProject&connectorRef=GitShivamtest&repoName=shivam-test&filePath=.harness%2FTestTemp1234_v1.yaml' \
--header 'Accept: */*' \
--header 'Accept-Language: en-GB,en-US;q=0.9,en;q=0.8' \
--header 'Authorization: Bearer eyJhbGciOiJIUzI1NiJ9.eyJhY2NvdW50SWQiOiJrbXB5U21VSVNpbW9SckpMNk5MNzN3IiwiYXV0aFRva2VuIjoickZ3ZkFuNWtyTDlRM3Rpd1RlUUJ2ZFRNU09YcVdsOGgiLCJpc3MiOiJIYXJuZXNzIEluYyIsIm5hbWUiOiJsdjBldVJoS1JDeWlYV3pTN3BPZzZnIiwidXNySWQiOiJsdjBldVJoS1JDeWlYV3pTN3BPZzZnIiwiZXhwIjoxNzAyMzYwMTQyLCJlbnYiOiIiLCJ0eXBlIjoiVVNFUiIsImlhdCI6MTcwMjI3Mzc0MiwiZW1haWwiOiJhZG1pbkBoYXJuZXNzLmlvIiwidXNlcm5hbWUiOiJBZG1pbiJ9.FEM6gPLLQb6FuVf-Z6rBSb5A9__JK-eJEtjKXoqe0iI' \
--header 'Connection: keep-alive' \
--header 'Cookie: jenkins-timestamper-offset=-19800000; __stripe_mid=e58251a3-2210-41f2-8657-486019b5e50d041a7b; _hjSessionUser_3650791=eyJpZCI6ImY1YjBiOTAwLWQzMjYtNWFiNC05N2YzLTdmNDVkZWEzNWRiMyIsImNyZWF0ZWQiOjE2OTcyMDQwNzE5NzUsImV4aXN0aW5nIjp0cnVlfQ==; intercom-id-iw4sgfzm=d2cdcdce-b415-40ce-8534-ed9e5d6360ef; intercom-device-id-iw4sgfzm=8877662e-f408-4b7c-8b91-244cbd3711a0; intercom-session-iw4sgfzm=; ajs_user_id=admin@harness.io; ajs_anonymous_id=352787ab-1ae5-44fa-ab37-881908de17bf; ADRUM=s=1702273743279&r=https%3A%2F%2Flocalhost%3A8181%2Fauth%2F%3Fhash%3D358787835' \
--header 'Origin: https://localhost:8181' \
--header 'Referer: https://localhost:8181/ng/account/kmpySmUISimoRrJL6NL73w/home/orgs/default/projects/TestProject/setup/resources/templates' \
--header 'Sec-Fetch-Dest: empty' \
--header 'Sec-Fetch-Mode: cors' \
--header 'Sec-Fetch-Site: same-origin' \
--header 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36' \
--header 'content-type: application/json' \
--header 'sec-ch-ua: "Google Chrome";v="119", "Chromium";v="119", "Not?A_Brand";v="24"' \
--header 'sec-ch-ua-mobile: ?0' \
--header 'sec-ch-ua-platform: "macOS"' \
--data-raw '[{"connectorRef":"GitShivamtest","repoName":"shivam-test","filePath":".harness/TestTemp1234_v1.yaml", "version": "v1"}]'`
Test Cases:

1. Verified the Multiple files are getting updated
2. Verified the error for invalid version

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56597)

### CODE REVIEW

**Comment by shivamnegi94** [comment]
trigger uts

File: template-service/contracts/src/main/java/io/harness/template/resources/NGTemplateResource.java [hunk 1]
```diff
@@ -704,4 +704,28 @@ ResponseDTO<PageResponse<EntitySetupUsageDTO>> listTemplateEntityUsage(
           NGCommonEntityConstants.VERSION_LABEL_KEY) String versionLabel,
       @Parameter(description = "This contains details of Git Entity like Git Branch info to be updated")
       TemplateUpdateGitMetadataRequest request);
+
+  @POST
+  @Path("/update/git-metadata/{templateIdentifier}")
+  @Timed
+  @ResponseMetered
+  @ApiOperation(value = "Update git metadata details for a remote template", nickname = "updateGitDetails")
```
Comment (utkarshchoubey-harness) [inline template-service/contracts/src/main/java/io/harness/template/resources/NGTemplateResource.java:712]: change description to mention it updates multiple versions in git, change nickname and operationId it is used to swagger generation, above api and this api have same nickname and operation Id and description

Comment (mohitg0795) [reply template-service/contracts/src/main/java/io/harness/template/resources/NGTemplateResource.java:712]: can't we directly add open api for it? Why do we need to add old api resources?

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56597)* status: 4 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`template-service/contracts/src/main/java/io/harness/template/resources/NGTemplateResource.java` line 712 at r1](https://reviewable.io/reviews/harness/harness-core/56597#gh-1422241885:-NlPU5d_4BOEVDmfCT60:b-o61l39) ([raw file](https://github.com/harness/harness-core/blob/7fc71df762a598a86969c8dad9723d5fb691f05d/template-service/contracts/src/main/java/io/harness/template/resources/NGTemplateResource.java#L712)):*
<details><summary><i>Previously, shivamnegi94 (Shivam Negi) wrote…</i></summary><blockquote>

I have updated the existing endpoint
</blockquote></details>

We cannot change existing endpoints. Its a backward incompatible change.

Also, lets add directly new open api endpoint instead of adding old style endpoints.

**Pull request closed without merging**